### ISSUE TITLE
[BUG]: Incorrect state when re-visiting an already completed exploration

### ISSUE THREAD

**Issue by AkashPaloju** [oppia/design-team#220]
### Describe the bug

When a user revisits an exploration after completing it, the exploration resumes from the previous checkpoint rather than starting from the beginning.

### URL of the page where the issue is observed.

https://www.oppiatestserver.org/explore/G8ZgSj1O4rjE?story_url_fragment=piece-of-cake&topic_url_fragment=fractions&classroom_url_fragment=math&node_id=node_1

### Steps To Reproduce

1. Signin
2. Go to any exploration. say [this](https://www.oppiatestserver.org/explore/G8ZgSj1O4rjE?story_url_fragment=piece-of-cake&topic_url_fragment=fractions&classroom_url_fragment=math&node_id=node_1)
3. Progress through the exploration until reaching a checkpoint (do not complete the exploration).
4. Visit another page.
5. Return to the previous exploration (the one you paused earlier) and complete it.
6. Return to any other page, like the learner dashboard.
7. Revisit the previous exploration that you completed earlier.

### Expected Behavior

Once an exploration is completed the learner will start the exploration from the beginning the next time they visit it.

### Screenshots/Videos

[Screencast from 24-06-24 03:11:17 PM IST.webm](https://github.com/oppia/design-team/assets/119884665/94851ed3-6ff8-48b4-9ce0-863b77d5ae71)

### What device are you using?

Desktop, Mobile

### Operating System

Linux, Android

### What browsers are you seeing the problem on?

Chrome, Firefox

### Tips for developers

Before addressing the bug, please identify which PR caused the issue (you can follow the steps [here](https://github.com/oppia/oppia/wiki/How-to-find-the-commit-which-introduced-a-bug)). If you identify the PR, comment on the issue with a link to it. If not, mention the commit hash of the oldest commit you saw the bug on (and the month and year it was made in).

Then, please leave a comment with details of the approach that you plan to take to fix the issue (see [example](https://github.com/oppia/oppia/issues/19157#issuecomment-1858788463)).

**Note:** If this is your first Oppia issue, please make sure to follow our guidelines for [choosing an issue](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#choosing-a-good-first-issue) and [setting things up](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up). You will also need to show a demo of the fix working correctly on your local machine. Thanks!

**Comment by jasmeen04** [comment]
I want to work on this issue. Please assign it to me.

**Comment by seanlip** [comment]
@jasmeen04 Per the guidance at https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#choosing-a-good-first-issue, please provide an explanation of what your PR will do (with names of files you're changing, what you plan to change in each file, etc.). If it looks good, we can assign you to this issue. Please also follow the other instructions on that wiki page if you have not yet done so.

I  also noticed that you have similar comments on other issues. Please make sure that any future comments you leave to "claim" an issue have the necessary substantiation as described above, otherwise it becomes spammy. Thanks!

**Comment by Nicolas-cst** [comment]
Hello, someone is working on this issue ? 
If not, I would like to follow the steps you described on your last message @seanlip.

**Comment by seanlip** [comment]
@Nicolas-cst Sure thing, please demonstrate this working correctly on your local machine and explain what changes you are making, as described in the previous comment. If it looks good, we can assign this issue to you. Thanks!

**Comment by Nicolas-cst** [comment]
Okay, I'll keep you posted on my progress.
Have a nice day !

**Comment by torack51** [comment]
Hi, 
My intuition is that the app doesn't correctly reset the user's progression of the exploration when the last card is reached. I noticed that when you jump back to a partly completed exploration, a modal appears offering to resume or to restart. Pressing the 'restart' button calls the function resetExplorationProgressAsync located here https://github.com/oppia/oppia/blob/6389baf1f2b65d01592c6fb48dd690fc1d227dee/core/templates/domain/exploration/editable-exploration-backend-api.service.ts#L222-L229
On the other hand, the function _addNewCard located here https://github.com/oppia/oppia/blob/6389baf1f2b65d01592c6fb48dd690fc1d227dee/core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts#L916 is called when we navigate to a new card, and a specific branch of the function gets called when the next card is the last card of the exploration https://github.com/oppia/oppia/blob/6389baf1f2b65d01592c6fb48dd690fc1d227dee/core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts#L956
I wanted to try to call resetExplorationProgressAsync when this specific branch is valid. It seems to be working locally as you can see on the video. It basically only implies adding the line `this.editableExplorationBackendApiService.resetExplorationProgressAsync(this.explorationId);` at the branch that I talked about earlier.
What do you think of this suggestion? Does it seem enough of a progress to assign me this issue? Also don't worry about Nicolas-cst, I'm his classmate and I work with him on the issue.

https://github.com/user-attachments/assets/e299c217-7e28-4d5b-abb5-04b91855d608

**Comment by seanlip** [comment]
Hi @torack51, thanks for looking into this and sorry for the late reply.

We've recently gotten some feedback suggesting that restarting from the beginning might not be the way to go ... we should offer users the option at least. See https://github.com/oppia/oppia/issues/21193 . So I think resetting their progress silently might not be the way to go...
 
Given that this looks like it needs some design work, I'm taking the "good first issue" label off it. Sorry about that -- I think we'll need to revisit these issues and figure out what the right experience is here first! Thank you and @Nicolas-cst for taking a look at it.

**Comment by yuanzhew-um** [comment]
@seanlip Hi, Is this issue still open? May I work on this issue if possible?

**Comment by seanlip** [comment]
@yuanzhew-um No, please don't, it's blocked on design as explained above. Please pick another issue -- thanks.

**Comment by EskAlam06** [comment]
this is still reproducible by following the steps -


1. Signin
2. Go to any exploration. say [this](https://www.oppiatestserver.org/explore/G8ZgSj1O4rjE?story_url_fragment=piece-of-cake&topic_url_fragment=fractions&classroom_url_fragment=math&node_id=node_1)
3. Progress through the exploration until reaching a checkpoint (do not complete the exploration).
4. Visit another page.
5. Return to the previous exploration (the one you paused earlier) and complete it.
6. Return to any other page, like the learner dashboard.
7. Revisit the previous exploration that you completed earlier.


Timestamp (1:40 - 2:34)-

https://github.com/user-attachments/assets/b7aa7677-6df8-4aba-b0af-1937592ba784
